src/relay/pass/fuse_ops.cc

```diff
@@ -261,9 +261,23 @@ class IndexedForwardGraph::Creator : private ExprVisitor {
   }
 
   void VisitExpr_(const TupleGetItemNode* op) final {
-    CHECK(graph_.node_map.count(op));
-    Node* node = graph_.node_map.at(op);
-    this->Update(op->tuple, node, kOpaque);
+    auto tuple_type = op->tuple->checked_type().as<TupleTypeNode>();
+    CHECK(tuple_type);
+    bool has_reference = false;
+    for (auto ty : tuple_type->fields) {
+      if (ty.as<RefTypeNode>()) {
```

Comment:
what if there is a reference inside? does it get recursively handled?

Reply:
Tuple fields are recursively visited via call to ExprVisitor::VisitExpr_(op) below. Ref nodes are never fused with its parent TupleGetItemNode.

Reply:
Please add the comment block to explain this so that future developers won't be confused here